File: apps/cs-master/lib/bris-parser/src/edu-detail.js
/**
 * 교육세부 (education_view.asp) HTML → 과정 상세 추출
 */

import { getDOMParser } from './dom.js';
import { normalizeText } from './utils.js';

export function extractEduDetailData(html) {
  const DOMParser = getDOMParser();
  const parser = new DOMParser();
  const doc = parser.parseFromString(html, 'text/html');

  const bizInput = doc.querySelector('input[id="businessId"], input[name="businessId"], input[name="business_id"]');
  const businessId = bizInput ? bizInput.value : '';

  let orderCode = '', projectId = '';
  const projectLink = doc.querySelector('a[href*="project_view.asp"]');
  if (projectLink) {
    orderCode = projectLink.textContent.trim();
    const pidMatch = (projectLink.getAttribute('href') || '').match(/PROJECT_ID=(\d+)/);
    projectId = pidMatch ? pidMatch[1] : '';
  }

  const tdHeaders = doc.querySelectorAll('td[bgcolor="#DDECB4"]');
  let company = '', internalManager = '', courseName = '', programName = '';
  let customerId = '', customerName = '';

  for (const th of tdHeaders) {
    const text = normalizeText(th.textContent).replace(/\*/g, '');
    const nextTd = th.nextElementSibling;
    if (!nextTd) continue;
    const val = normalizeText(nextTd.textContent);
    if (text === '회사명') company = val;
    else if (text === '담당자') {
      internalManager = val;
      customerId = nextTd.getAttribute('data-customer-id') || '';
      customerName = nextTd.getAttribute('data-customer-name') || '';
      if (!customerId) {
        const dmLink = nextTd.querySelector('a[href*="dm_view.asp"], a[href*="CUSTOMER_ID"]');
        if (dmLink) {
          const cidMatch = (dmLink.getAttribute('href') || '').match(/CUSTOMER_ID=(\d+)/);
          if (cidMatch) customerId = cidMatch[1];
          if (!customerName) customerName = dmLink.textContent.trim();
        }
      }
    }
    else if (text === '과정명') courseName = val;
    else if (text === '프로그램명') programName = val;
  }

  // fallback: 페이지 전체에서 dm_view.asp 링크 탐색
  if (!customerId) {
    const dmLinkFallback = doc.querySelector('a[href*="dm_view.asp"]');
    if (dmLinkFallback) {
      const cidMatch = (dmLinkFallback.getAttribute('href') || '').match(/CUSTOMER_ID=(\d+)/);
      if (cidMatch) customerId = cidMatch[1];
      if (!customerName) customerName = dmLinkFallback.textContent.trim();
    }
  }

  // 강사: inst_fee_edit 링크의 텍스트
  const instructorLinks = doc.querySelectorAll('a[href*="inst_fee_edit.asp"]');
  const instructors = [...instructorLinks].map(a => a.textContent.trim()).filter(Boolean);

  // 진행자(LF): al_person 배열에서 name 추출 (raw HTML에서 직접 매칭)
  let facilitators = [];
  const alMatch = html.match(/al_person\s*:\s*\[([\s\S]*?)\]/);
  if (alMatch && alMatch[1].trim()) {
    try {
      const fixed = alMatch[1].replace(/(\w+)\s*:/g, '"$1":').replace(/'/g, '"');
      const arr = JSON.parse('[' + fixed + ']');
      facilitators = [...new Set(arr.map(p => p.name).filter(Boolean))];
    } catch(e) {
      const nameMatches = alMatch[1].match(/name\s*:\s*['"]([^'"]+)['"]/g);
      if (nameMatches) {
        facilitators = [...new Set(nameMatches.map(m => {
          const v = m.match(/['"]([^'"]+)['"]\s*$/);
          return v ? v[1] : '';
        }).filter(Boolean))];
      }
    }
  }

  // 대면/비대면
  let eduDelivery = '';
  for (const th of tdHeaders) {
    const text = th.textContent.trim();
    if (text.includes('비대면')) {
      const nextTd = th.nextElementSibling;
      if (nextTd) {
        const val = nextTd.textContent.trim();
        eduDelivery = val.includes('비대면') ? '비대면' : val.includes('대면') ? '대면' : val;
      }
      break;
    }
  }

  return {
    businessId, orderCode, projectId, company, internalManager,
    courseName, programName, customerId, customerName,
    instructors, facilitators, eduDelivery
  };
}

File: apps/cs-master/lib/bris-parser-py/src/bris_parser/edu_detail.py
"""
교육세부 (education_view.asp) HTML → 과정 상세 추출
JS lib/bris-parser/src/edu-detail.js 와 1:1 동등.
"""

import json
import re
from bs4 import BeautifulSoup

from .utils import normalize_text


def extract_edu_detail_data(html: str) -> dict:
    soup = BeautifulSoup(html, 'html.parser')

    biz_input = soup.select_one(
        'input[id="businessId"], input[name="businessId"], input[name="business_id"]'
    )
    business_id = biz_input.get('value', '') if biz_input else ''

    order_code = ''
    project_id = ''
    project_link = soup.select_one('a[href*="project_view.asp"]')
    if project_link:
        order_code = project_link.get_text(strip=True)
        href = project_link.get('href', '') or ''
        m = re.search(r'PROJECT_ID=(\d+)', href)
        project_id = m.group(1) if m else ''

    td_headers = soup.select('td[bgcolor="#DDECB4"]')
    company = ''
    internal_manager = ''
    course_name = ''
    program_name = ''
    customer_id = ''
    customer_name = ''

    for th in td_headers:
        text = normalize_text(th.get_text()).replace('*', '')
        next_td = th.find_next_sibling()
        if not next_td:
            continue
        val = normalize_text(next_td.get_text())
        if text == '회사명':
            company = val
        elif text == '담당자':
            internal_manager = val
            customer_id = next_td.get('data-customer-id', '') or ''
            customer_name = next_td.get('data-customer-name', '') or ''
            if not customer_id:
                dm_link = next_td.select_one(
                    'a[href*="dm_view.asp"], a[href*="CUSTOMER_ID"]'
                )
                if dm_link:
                    href = dm_link.get('href', '') or ''
                    m = re.search(r'CUSTOMER_ID=(\d+)', href)
                    if m:
                        customer_id = m.group(1)
                    if not customer_name:
                        customer_name = dm_link.get_text(strip=True)
        elif text == '과정명':
            course_name = val
        elif text == '프로그램명':
            program_name = val

    # fallback: 페이지 전체에서 dm_view.asp 링크 탐색
    if not customer_id:
        dm_link_fallback = soup.select_one('a[href*="dm_view.asp"]')
        if dm_link_fallback:
            href = dm_link_fallback.get('href', '') or ''
            m = re.search(r'CUSTOMER_ID=(\d+)', href)
            if m:
                customer_id = m.group(1)
            if not customer_name:
                customer_name = dm_link_fallback.get_text(strip=True)

    # 강사: inst_fee_edit 링크 텍스트
    instructor_links = soup.select('a[href*="inst_fee_edit.asp"]')
    instructors = [
        a.get_text(strip=True) for a in instructor_links if a.get_text(strip=True)
    ]

    # 진행자(LF): al_person 배열에서 name 추출
    facilitators = []
    al_match = re.search(r'al_person\s*:\s*\[([\s\S]*?)\]', html)
    if al_match and al_match.group(1).strip():
        try:
            fixed = re.sub(r'(\w+)\s*:', r'"\1":', al_match.group(1))
            fixed = fixed.replace("'", '"')
            arr = json.loads('[' + fixed + ']')
            seen = set()
            for p in arr:
                nm = p.get('name') if isinstance(p, dict) else None
                if nm and nm not in seen:
                    seen.add(nm)
                    facilitators.append(nm)
        except Exception:
            # 패턴 2: name 필드만 regex 추출
            seen = set()
            for m in re.finditer(
                r'name\s*:\s*[\'"]([^\'"]+)[\'"]', al_match.group(1)
            ):
                nm = m.group(1)
                if nm and nm not in seen:
                    seen.add(nm)
                    facilitators.append(nm)

    # 대면/비대면
    edu_delivery = ''
    for th in td_headers:
        text = th.get_text(strip=True)
        if '비대면' in text:
            next_td = th.find_next_sibling()
            if next_td:
                val = next_td.get_text(strip=True)
                if '비대면' in val:
                    edu_delivery = '비대면'
                elif '대면' in val:
                    edu_delivery = '대면'
                else:
                    edu_delivery = val
            break

    return {
        'businessId': business_id,
        'orderCode': order_code,
        'projectId': project_id,
        'company': company,
        'internalManager': internal_manager,
        'courseName': course_name,
        'programName': program_name,
        'customerId': customer_id,
        'customerName': customer_name,
        'instructors': instructors,
        'facilitators': facilitators,
        'eduDelivery': edu_delivery,
    }
